plugins/inputs/ecs/ecs.go

```diff
@@ -0,0 +1,251 @@
+
+  ## Set to true to configure from env vars
+  envcfg = false
+
+  ## Containers to include and exclude. Globs accepted.
+  ## Note that an empty array for both will include all containers
+  container_name_include = []
+  container_name_exclude = []
+
+  ## Container states to include and exclude. Globs accepted.
+  ## When empty only containers in the "running" state will be captured.
+  # container_status_include = []
+  # container_status_exclude = []
+
+  ## ecs labels to include and exclude as tags.  Globs accepted.
+  ## Note that an empty array for both will include all labels as tags
+  ecs_label_include = [ "com.amazonaws.ecs.*" ]
+  ecs_label_exclude = []
+
+  ## Timeout for docker list, info, and stats commands
+  timeout = "5s"
+`
+
+// Description describes ECS plugin
+func (ecs *Ecs) Description() string {
+	return "Read metrics about docker containers from Fargate/ECS v2 meta endpoints."
+}
+
+// SampleConfig returns the ECS example config
+func (ecs *Ecs) SampleConfig() string {
+	return sampleConfig
+}
+
+// Gather is the entrypoint for telegraf metrics collection
+func (ecs *Ecs) Gather(acc telegraf.Accumulator) error {
+	err := initSetup(ecs)
+	if err != nil {
+		return err
+	}
+
+	task, stats, err := PollSync(ecs.client)
+	if err != nil {
+		return err
+	}
+
+	mergeTaskStats(task, stats)
+
+	taskTags := map[string]string{
+		"cluster":  task.Cluster,
+		"task_arn": task.TaskARN,
+		"family":   task.Family,
+		"revision": task.Revision,
+	}
+
+	// accumulate metrics
+	ecs.accTask(task, taskTags, acc)
+	ecs.accContainers(task, taskTags, acc)
+
+	return nil
+}
+
+func initSetup(ecs *Ecs) error {
+	if ecs.client == nil {
+		var c *EcsClient
+		var err error
+		if ecs.EnvCfg {
+			c, err = ecs.newEnvClient()
+		} else {
+			c, err = ecs.newClient(ecs.Timeout.Duration)
+		}
+		if err != nil {
+			return err
+		}
+
+		c.BaseURL, err = url.Parse(ecs.EndpointURL)
+		if err != nil {
+			return err
+		}
+
+		ecs.client = c
+	}
+
+	// Create filters
+	if !ecs.filtersCreated {
+		err := ecs.createContainerNameFilters()
+		if err != nil {
+			return err
+		}
+		err = ecs.createContainerStatusFilters()
+		if err != nil {
+			return err
+		}
+		err = ecs.createLabelFilters()
+		if err != nil {
+			return err
+		}
+		ecs.filtersCreated = true
+	}
+
+	return nil
+}
+
+func (ecs *Ecs) accTask(task *Task, tags map[string]string, acc telegraf.Accumulator) {
+	taskFields := map[string]interface{}{
+		"revision":       task.Revision,
+		"desired_status": task.DesiredStatus,
+		"known_status":   task.KnownStatus,
+		"limit_cpu":      task.Limits["CPU"],
+		"limit_mem":      task.Limits["Memory"],
+	}
+
+	acc.AddFields("ecs_task", taskFields, tags, task.PullStoppedAt)
+}
+
+func (ecs *Ecs) accContainers(task *Task, taskTags map[string]string, acc telegraf.Accumulator) {
+	for _, c := range task.Containers {
+		if !ecs.containerNameFilter.Match(c.Name) {
+			log.Printf("container %v did not match name filter", c.ID)
```

Comment:
I would probably remove this log because it is fairly easy to test if a container was filtered by modifying the filter and it is likely to be quite verbose when used.  If you really want to keep it then emit it as a debug message:
```
log.Printf("D! [inputs.ecs] Container %v did not match name filter"
```